src/RevSysCalls.cc

```diff
@@ -16,41 +16,44 @@ EcallStatus RevProc::EcallLoadAndParseString(RevInst& inst,
-  // at a time and search for the string terminator character '\0'
-  if(EcallState.bytesRead != 0){
-    EcallState.string += std::string_view(EcallState.buf.data(), EcallState.bytesRead);
-    EcallState.bytesRead = 0;
-  }
-
-  // We store the 0-terminator byte in EcallState.string to distinguish an empty
-  // C string from no data read at all. If we read an empty string in the
-  // program, EcallState.string.size() == 1 with front() == back() == '\0'. If no
-  // data has been read yet, EcallState.string.size() == 0.
-  if(EcallState.string.size() && !EcallState.string.back()){
-    //found the null terminator - we're done
-    // action is usually passed in as a lambda with local code and captures
-    // from the caller, such as performing a syscall using EcallState.string.
-    action();
-
-    EcallState.string.clear();   //reset the ECALL buffers
-    EcallState.bytesRead = 0;
-
-    DependencyClear(HartToExec, 10, false);
-    rtval = EcallStatus::SUCCESS;
-  }else{
-    //We are in the middle of the string - read one byte
-    MemReq req{straddr + EcallState.string.size(), 10, RevRegClass::RegGPR, HartToExec, MemOp::MemOpREAD, true, [=](const MemReq& req){this->MarkLoadComplete(req);}};
-    LSQueue->insert({make_lsq_hash(req.DestReg, req.RegType, req.Hart), req});
-    mem->ReadVal(HartToExec,
-                 straddr + EcallState.string.size(),
-                 EcallState.buf.data(),
-                 req,
-                 REVMEM_FLAGS(0));
-    EcallState.bytesRead = 1;
-    std::cout << "Setting dependency for HartToExec == " << HartToExec << std::endl;
-    DependencySet(HartToExec, 10, false);
+  if( RegFile->GetLSQueue()->count(make_lsq_hash(10, RevRegClass::RegGPR, HartToExec)) > 0 ){
     rtval = EcallStatus::CONTINUE;
+  } else {
+    // we don't know how long the path string is so read a byte (char)
+    // at a time and search for the string terminator character '\0'
+    if(EcallState.bytesRead != 0){
+      EcallState.string += std::string_view(EcallState.buf.data(), EcallState.bytesRead);
+      EcallState.bytesRead = 0;
+    }
+
+    // We store the 0-terminator byte in EcallState.string to distinguish an empty
+    // C string from no data read at all. If we read an empty string in the
+    // program, EcallState.string.size() == 1 with front() == back() == '\0'. If no
+    // data has been read yet, EcallState.string.size() == 0.
+    if(EcallState.string.size() && !EcallState.string.back()){
+      //found the null terminator - we're done
+      // action is usually passed in as a lambda with local code and captures
+      // from the caller, such as performing a syscall using EcallState.string.
+      action();
+
+      EcallState.string.clear();   //reset the ECALL buffers
+      EcallState.bytesRead = 0;
+
+      DependencyClear(HartToExec, 10, false);
+      rtval = EcallStatus::SUCCESS;
+    }else{
+      //We are in the middle of the string - read one byte
```

Comment:
you guys really need to start paying attention to the 80 character limits.  Code like this is illegible on anything except VSCode with a 100" monitor

Reply:
I generally try to keep lines under ~80-100 characters.

One tip: if a line is too long because it has a long string, the string can be split into adjacent strings, like:
```c++
"very long strings "
"can be split across lines"
```
I usually put the separating space at the ends of the split string(s).

I could make a pass at re-wrapping the lines intelligently with `emacs`, or more fancy with `clang-format`, but it would be disruptive and cause merge conflicts with orhers' work in progress, so it would need to done at a strategic time, such as after a deliverable.

In rocBLAS we used a pre-commit script which called `clang-format` on the committed file, and rejected the commit if `clang-format` produced differences. I could develop a similar script for Rev. It would incrementally reformat the source files at each commit, rather than reformatting the whole source tree. I will create a `clang-format` file which keeps most of the style used today.

Reply:
I use neovim and have a 27" monitor 😭... I'll go ahead and make the lines 80 wide

Reply:
Alright all the code lines are now no longer than 80-90 chars long